smallab/checkpointed_experiment_handler.py

```diff
@@ -0,0 +1,64 @@
+import datetime
+import logging
+import pickle
+import typing
+
+import os
+from dateutil.parser import parse
+
+from smallab.experiment import CheckpointedExperiment
+# TODO this should handle serializing a checkpointable experiment
+from smallab.file_locations import get_partial_save_directory
+from smallab.specification_hashing import specification_hash
+from smallab.types import Specification
+
+
+class CheckpointedExperimentHandler():
+    """
+    An internal handler to handle running checkpointed experiments.
+    """
+    def __init__(self, rolled_backups=3):
+        self.rolled_backups = rolled_backups
```

Comment:
Maybe add a docstring describing what `rolled_backups` does?

Reply:
as of right now there's no way of an end user configuring this, but i'll add docs just in case